src/librustc/ty/maps.rs

```diff
@@ -804,6 +845,10 @@ fn metadata_dep_node(def_id: DefId) -> DepNode<DefId> {
     DepNode::MetaData(def_id)
 }
 
+fn cratenum_metadata_dep_node(crate_num: CrateNum) -> DepNode<DefId> {
+    DepNode::MetaDataByCrateNum(crate_num)
```

Comment:
You can just use `MetaDataByCrateNum` above, the functions are used when the variants don't match the key exactly.